chain-tx-enclave-next/mls/src/astree.rs

```diff
@@ -17,11 +17,9 @@ impl<D: Input + BlockInput + FixedOutput + Reset + Default + Clone> TreeBaseKeyS
     pub fn new(application_secret: SecretVec<u8>, leaf_count: LeafSize) -> Self {
         let root_index = NodeSize::root(leaf_count);
         let node_len = NodeSize::node_width(leaf_count);
-        let mut secrets = Vec::with_capacity(node_len.0);
-        for _ in 0..node_len.0 {
-            secrets.push(None);
-        }
-        secrets[root_index.0] = Some(application_secret);
+        let mut secrets = Vec::new();
```

Comment:
how about
vec![...]. instead of resize_with

Reply:
Oh, it can't do that because the secret type don't implement `Clone`.

Reply:
or `Vec::with_capacity`?